server/src/main/java/org/folio/shared/index/storage/Storage.java

```diff
@@ -92,8 +97,25 @@ public Storage(RoutingContext routingContext) {
                 + ")",
             "CREATE UNIQUE INDEX IF NOT EXISTS match_key_value_idx ON " + matchKeyValueTable
                 + " (match_key_config_id, match_value, bib_record_id)",
-            "CREATE INDEX IF NOT EXISTS match_key_value_bib_id_idX ON " + matchKeyValueTable
-                + " (bib_record_id)"
+            "CREATE INDEX IF NOT EXISTS match_key_value_bib_id_idx ON " + matchKeyValueTable
+                + " (bib_record_id)",
+            CREATE_IF_NO_EXISTS + clusterRecordTable
+                + "(record_id uuid NOT NULL,"
+                + " match_key_config_id VARCHAR NOT NULL,"
+                + " cluster_id uuid NOT NULL,"
+                + " FOREIGN KEY(record_id) REFERENCES " + bibRecordTable + " ON DELETE CASCADE)",
```

Comment:
Would it also make sense to add FK on match_key_config_id to remove all clusters when corresponding matchkey configuration is removed?

Reply:
Sure. I'll try that.. This will save the manual removal when a match key configuration is delete.

Reply:
Done now.